src/generators/rust/renderers/PackageRenderer.ts

```diff
@@ -0,0 +1,62 @@
+import { RustRenderer } from '../RustRenderer';
+import { PackagePresetType } from '../RustPreset';
+import { ConstrainedMetaModel, MetaModel } from '../../../models';
+import { RustOptions } from '../RustGenerator';
+import { FormatHelpers } from '../../../helpers';
+import { defaultModelNameConstraints } from '../constrainer/ModelNameConstrainer';
+/**
+ * Renderer for Rust's supporting files
+ * 
+ * @extends RustRenderer
+ */
+export class PackageRenderer extends RustRenderer<ConstrainedMetaModel> {
+  public defaultSelf(): string {
+    return '';
+  }
+}
+
+export const RUST_DEFAULT_PACKAGE_PRESET: PackagePresetType<RustOptions> = {
+  self({ renderer }) {
+    return renderer.defaultSelf();
+  },
+
+  lib({ inputModel, renderer }) {
+    const modelNames: string[] = Object.values(inputModel.models).map((m: MetaModel) => m.name);
+    const imports = renderer.renderBlock(modelNames.map(modelName => {
+      let mod = defaultModelNameConstraints()({ modelName });
+      mod = FormatHelpers.snakeCase(mod);
+      return `
+pub mod ${mod};
+pub use self::${mod}::*;`;
+    }).flat());
+
+    return `#[macro_use]
+extern crate serde;
+extern crate serde_json;
```

Comment:
Think my lack of Rust knowledge comes to play here and why I did not notice this before 😅 I think it's fine to keep it for now, I just want to make you aware of this 🙂 

I expect that in the future you will get request to remove this from the standard rendering, and instead move this to a `SerdePreset`. Don't have a full overview of whether this is even possible, depends on how tangled things are with each other I guess.

The reason is, generally, renderers just render the very basic model, with "no" dependencies and are the raw building blocks presets can extend i.e. with serde annotation.

This also enables us to all use the common renderers and then decide to add extra sprinkles on top, i.e. libraries to help serialize the models to JSON, XML or even bytes, or something completely different 🙂 

Similar raised issue for Java: https://github.com/asyncapi/modelina/issues/812

Reply:
Moving this to a preset in a later revision makes sense to me! For my own application, I have to support 2 ser/de strategies:

* https://crates.io/crates/serde_bytes for the raw output of an edge ML model
* https://docs.rs/serde_json/latest/serde_json/ for human-readable data

I'd like to put together 1 concrete example of a JSON-based NATs Rust application generated from an AsyncAPI schema, then circle back to the more abstract parts (like Serde Presets). 

There's quite a bit to chew on here when considering the transports! For example...

Rust's `nats.rs` crate brokers `bytes::Byte`, which is a low-level container for handling contiguous bits of memory (exactly what I expect from a transport layer). A `From<bytes::Byte`> implementation is required to serialize/deserialize this data structure into a model struct/enum/etc. 
https://github.com/nats-io/nats.rs/blob/main/async-nats/src/message.rs#L28

However, the `jetstream` enabled Nats.rs client does implement Serializable/Deserializble traits, plus the message body is a primitive u64 sequence (not `bytes::Byte` wrapper) https://github.com/nats-io/nats.rs/blob/main/async-nats/src/jetstream/stream.rs#L558

tldr: I'm still trying to figure out exactly where the boundaries for the Rust generator will be. I'm seeing a lot of possible permutations (below), but am just working on the `rust-nats-serde-json` story to simplify my path forward. 

* rust-nats-serde-<json|bytes|etc>
* rust-nats-jetstream-<json|bytes|etc>
* rust-nats-raw (no serde)

:open_mouth: :exploding_head:

Reply:
Definitely many options 😄 

Makes sense to me 👍